generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/VersionResolver.java

```diff
@@ -0,0 +1,109 @@
+package com.google.devtools.build.workspace.maven;
+
+
+import com.google.common.collect.Lists;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.VersionRangeRequest;
+import org.eclipse.aether.resolution.VersionRangeResolutionException;
+import org.eclipse.aether.resolution.VersionRangeResult;
+
+import java.util.List;
+
+import static com.google.devtools.build.workspace.maven.ArtifactBuilder.InvalidArtifactCoordinateException;
+
+/**
+ * Given a Maven coordinate with a version specification resolves the version of the coordinate in a
+ * similar fashion as Maven. Version specifications can include hard and soft pins as well as various
+ * forms of version ranges. When given a version range, Maven selects the highest available version.
+ * For both soft and hard pins, e.g. "4.2" or "[4.2]", Maven defaults to the pinned version.
+ */
+class VersionResolver {
+
+  private final List<RemoteRepository> remoteRepositories;
+  private final RepositorySystem repositorySystem;
+  private final RepositorySystemSession repositorySystemSession;
+
+  private VersionResolver(RepositorySystem repositorySystem,
```

Comment:
Make AetherUtils non-static, pass it in here, and add helper methods when you need to contact aether (`aetherUtils.requestVersionRange()`).  Then you can customize repos & test.